openstack/cinder/templates/vcenter_datacenter_cinder_configmap.yaml

```diff
@@ -48,6 +48,7 @@ template: |
       vmware_storage_profile = {{ .Values.backends.standard_hdd.vmware_storage_profile }}
       extra_capabilities = '{"vcenter-shard": "{= host.split(".")[0] =}"}'
 
+      {{ .Values.backends.custom_backend }}
```

Comment:
How about a different structure, that would allow more backends. I haven't tested it but this code might work
```
{{- range $backend, $kvs := .Values.backends.additional_backends }}
[{{ $backend }}]
{{- range $key, $value := $kvs }}
{{ $key }} = {{ $value }}
{{- end }}
{{- end }}
```
and would look like this in the region's YAML:
```yaml
backends:
  additional_backends:
    pure_vmware:
      volume_driver: cinder.volume.drivers.vmware.vmdk.VMwareVcVmdkDriver
      vmware_storage_profile: cinder-pure
      extra_capabilities: '{"vcenter-shard": "{= host.split(".")[0] =}"}'
```

Reply:
The custom_backend var as I have in this patch can be more than 1 backend, since it's a free form field.

```
backends:
custom_backend: |
[custom_backend_name]
volume_backend_name = my_custom_backend
volume_driver = cinder.volume.drivers.something.custom

[vmware_pure]
volume_backend_name = vmware_pure
volume_driver = cinder.volume.drivers.blah
vcenter-shard = something

Reply:
free-form fields have the disadvantage for me, that they are not YAML and thus not integrate so well into the rest of the config and it's often harder to spot errors in there.

But you're right. To indicate that, maybe we should rename it to `custom_backends`?